lib/Language/Slicer/Slice.hs

```diff
@@ -244,5 +249,16 @@ bwdSliceM outcome trace = do
                rho2' = unbindEnv  rho2 x
            (rho1, e1, t1') <- bwdSliceM (OExn p1) t1
            return (rho1 `lub` rho2', ETryWith e1 x e2, TTryWith t1' x t2')
+
+    -- JSTOLAREK: hacking OHole rules
```

Comment:
Yes, I'll take a closer look at that.  That's one reason I wanted to merge first so that I don't make changes that conflict with the concurrent changes to Slice.hs.  I'll merge once the tests pass.